File: BOJ(Java 공부용)/1260.java
import java.io.*;
import java.lang.reflect.Array;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);
        int st = Integer.parseInt(input[2]);
        ArrayList<Integer>[] graph = new ArrayList[n+1];
        boolean visited[] = new boolean[n+1];
        for(int i=1;i<=n;i++) {
            graph[i] = new ArrayList<>();
        }
        for(int i=0;i<m;i++) {
            String[] s = br.readLine().split(" ");
            int here = Integer.parseInt(s[0]);
            int next = Integer.parseInt(s[1]);
            graph[here].add(next);
            graph[next].add(here);
        }
        for(int i=1;i<=n;i++) {
            Collections.sort(graph[i]);
        }
        dfs(st,graph,visited);
        System.out.println();
        visited = new boolean[n+1];
        bfs(st,graph,visited);
        bw.close();
    }
    private static void dfs(int here, ArrayList<Integer>[] graph, boolean[] visited) throws IOException {
        visited[here] = true;
        bw.write(String.valueOf(here)+" ");
        bw.flush();
        for(int i=0;i<graph[here].size();i++) {
            int next = graph[here].get(i);
            if(!visited[next]) dfs(next,graph,visited);
        }
    }
    private static void bfs(int here,ArrayList<Integer>[] graph,boolean[] visited) throws IOException {
        ArrayDeque<Integer> q = new ArrayDeque<>();
        q.add(here);
        visited[here] = true;
        while(!q.isEmpty()) {
            int Here = q.peekFirst();
            bw.write(String.valueOf(Here)+" ");
            bw.flush();
            q.pop();
            for(int i=0;i<graph[Here].size();i++){
                int next = graph[Here].get(i);
                if(!visited[next]) {
                    visited[next] = true;
                    q.add(next);
                }
            }
        }
     }
}

File: BOJ(Kotlin)/1260.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

val bw = BufferedWriter(OutputStreamWriter(System.out))
fun main(args : Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))

    val sizeList = br.readLine().split(" ")
    val n = sizeList.get(0).toInt()
    val m = sizeList.get(1).toInt()
    val st = sizeList.get(2).toInt()
    var graph  = Array(n+1){ mutableListOf<Int>()}
    var visited = BooleanArray(n+1)
    for (i in 1..m) {
        val s = br.readLine().split(" ")
        val here = s.get(0).toInt()
        val next = s.get(1).toInt()
        graph[here].add(next)
        graph[next].add(here)
    }

    for (i in 1..n) {
        graph[i].sort()
    }

    dfs(st,graph,visited)
    bw.write("\n")
    visited = BooleanArray(n+1)
    bfs(st,graph,visited)
    bw.flush()
    bw.close()
}

fun dfs(here : Int, graph : Array<MutableList<Int>>,visited : BooleanArray ) {
    visited[here] = true
    bw.write(here.toString()+" ")
    for(i in 0 until graph[here].size) {
        val next = graph[here].get(i)
        if (!visited[next]) {
            dfs(next,graph,visited)
        }
    }
}

fun bfs(here : Int, graph : Array<MutableList<Int>>, visited : BooleanArray) {
    val q = ArrayDeque<Int>()
    q.add(here)
    visited[here] = true
    while(!q.isEmpty()) {
        val here = q.first()
        bw.write(here.toString()+" ")
        q.pollFirst()
        for (i in 0 until graph[here].size) {
            val next = graph[here].get(i)
            if (!visited[next]) {
                visited[next] = true
                q.add(next)
            }
        }
    }
}